binding-library/java/src/main/java/com/microsoft/azure/functions/signalr/annotation/SignalRConnectionInfoInput.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.functions.signalr.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.ElementType;
+
+import com.microsoft.azure.functions.annotation.CustomBinding;
+
+/**
+ * <p>Place this on a parameter to obtain a SignalRConnectionInfo object.
+ * The parameter type can be one of the following:</p>
+ *
+ * <ul>
+ *     <li>SignalRConnectionInfo type</li>
+ * </ul>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@CustomBinding(direction = "in", name = "connectionInfo", type = "SignalRConnectionInfo")
+public @interface SignalRConnectionInfoInput {
+
+    /**
+     * The variable name used in function.json.
+     * @return The variable name used in function.json.
+     */
+    String name() default "";
```

Comment:
Name is required for an input binding (no default).

(It can be optional for output because it defaults to `$return` if used on a method)